components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/sequence/impl/GraphBasedSequenceHandler.java

```diff
@@ -144,6 +158,90 @@ private void displayLongWait(AuthenticationContext context, HttpServletRequest r
         }
     }
 
+    /**
+     * Handles the prompt.
+     *
+     * @param request
+     * @param response
+     * @param context
+     * @param sequenceConfig
+     * @param promptNode
+     * @return true if the execution needs to be stopped and show somethin to the user.
+     */
+    private boolean handlePrompt(HttpServletRequest request, HttpServletResponse response,
+                                 AuthenticationContext context, SequenceConfig sequenceConfig,
+                                 ShowPromptNode promptNode) throws FrameworkException {
+
+        boolean isPromptToBeDisplayed = false;
+        if (context.isReturning()) {
+            String action = "Success";
+            for (String s : request.getParameterMap().keySet()) {
+                if (s.startsWith("action.")) {
+                    action = s.substring("action.".length(), s.length());
+                    break;
+                }
+            }
+            action = "on" + action;
+            executeFunction(action, promptNode, context);
+            AuthGraphNode nextNode = promptNode.getDefaultEdge();
+            context.setProperty(FrameworkConstants.JSAttributes.PROP_CURRENT_NODE, nextNode);
+            context.setReturning(false);
+        } else {
+            displayPrompt(context, request, response, promptNode.getTemplateId(), promptNode.getData());
+            isPromptToBeDisplayed = true;
+        }
+        return isPromptToBeDisplayed;
+    }
+
+    private void displayPrompt(AuthenticationContext context, HttpServletRequest request, HttpServletResponse response,
+                               String templateId, String data) throws FrameworkException {
+
+        String promptId = UUID.randomUUID().toString();
+
+        try {
+            String promptPage = ConfigurationFacade.getInstance().getAuthenticationEndpointPromptURL();
+            String redirectUrl = promptPage + "?templateId=" + templateId + "&promptId=" + promptId;
```

Comment:
Need to sanitize "templateId" and "promptId" as they are user inputs from script.